### PR TITLE
fix: tools management

### PR DESCRIPTION
**Pull request by mmorel-35**
**What type of PR is this?**
* fix OSSF Score 

**What this PR does / why we need it**:
Today the tools are managed by a dedicated go.mod file .
The problem is that scorecard consider tools as dependencies of the project and each vulnerability of those tools decrease the score of the project.

This store tool and its version in a txt file to be then `go install`ed in `tools/bin`.
It also adds an actions/cache so it is cached to not download it to much time.

**Which issue(s) this PR fixes**:

Fixes most of the dependabot vulnerabilities

### CODE REVIEW

**Comment by mmorel-35** [comment]
@phlax , what do you think ?

File: tools/src/buf.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+github.com/bufbuild/buf/cmd/buf@v1.28.1
```
Comment (zirain) [inline tools/src/buf.txt:1]: maybe we can remove these file, define thing directly in makefile, wdyt @arkodg ?

**Comment by arkodg** [comment]
@mmorel-35 thanks for working on it, and eliminating the false positives to bump up the scorecard score
can you try out the approach suggested by @zirain to collapse it all under one file - `tools.mk` ?

**Comment by mmorel-35** [comment]
Hi ! I’ve merged install of the tools in tools.mk.
I don’t kown if there is a more elegant way to do this. I’m interested in any way to make this more simple.

Dependabot is not able to handle go modules outside of a go.mod file . I don’t know how it would be possible to use a bot to update those dependencies.
@arkodg , @zirain , @phlax ,

Any suggestion ?

**Comment by arkodg** [comment]
hey @mmorel-35 if dependabot cannot handle updates outside `go.mod`, imo we should stick with the orig `go.mod` approach, there's a big benefit of auto updating these tools that are used in the CI workflow

maybe we need to manually bump those tool dependencies for now

**Review by zirain — APPROVED** [review]
this's fine to me, a stable version of tools always lead a stable result, we can upgrade these manually if needed.

**Comment by zirain** [comment]
kindly ping @arkodg

**Comment by arkodg** [comment]
ah yah, please give me a day or two to try it out manually, else we can merge this PR

**Comment by arkodg** [comment]
looks like manually bumping the packages helped, this is no longer needed, thanks @mmorel-35 for trying this out

**Pull request closed without merging**